### ISSUE TITLE
Allow to select source and target branch *before* creating diff for the pull request

### ISSUE THREAD

**Issue by otbutz** [go-gitea/gitea#6900]
- Gitea version (or commit ref): 1.8.0
- Git version: 2.21.0
- Operating system: Ubuntu 18.04
- Database (use `[x]`):
  - [X] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite

## Description
The current behavior of the new pull request page is to immediately populate its content once the selection of either target or source branch is changed.

This is problematic for big repositories with long running support branches. Creating a pull request to fix a bug in our oldstable release would preselect `master...master` which is perfectly fine but as soon as i change either the target to our oldstable branch or the source to my bugfix branch, the page loads forever because our `master` and the oldstable branch differ by \~1k commits.

It would be nice if I could select both branches and start the diff only after pressing a button. If the URL in the address bar would change before trying to create the diff, that would also be a solution.

**Comment by m3rtl** [comment]
Are there any updates on this?
That would be really helpful.

**Comment by lunny** [comment]
I think a solution can be https://github.com/go-gitea/gitea/issues/13949#issuecomment-2495060316
